### DISCUSSION TITLE
Customize Endpoint - include columns in RetrieveRequest

### DISCUSSION THREAD

**Discussion by teafmartins** [serenity-is/Serenity#7151] [Help]
Hello,

I needed to customize the “Retrieve” endpoint so that it calls an external API. However, the external API to be called must pass in the URL the entityid and two more columns from the ROW, however the request only includes the EntityId, how can I include more fields?

        [HttpPost]
        public RetrieveResponse<MyRow> Retrieve(IDbConnection connection, RetrieveRequest request)
        {
            RetrieveResponse<MyRow> listReturn = new RetrieveResponse<MyRow>();

            var url = MyProject.GetURL(request.COLUMN);

            if (url != "")
            {
                using (var httpClient = new HttpClient())
                {
                    var response = httpClient.GetStringAsync(new Uri(url)).Result;
                    var j = JsonConvert.DeserializeObject<List<MyRow>>(response);
                    listReturn.Entity = j[0];
                }
            }
            return listReturn;
        }

![image](https://github.com/serenity-is/Serenity/assets/17210232/26a01baf-ca27-4a8e-b39a-639428f57c75)

**Comment by VictorTomaili** [comment]
??

```cs
public class MyRetrieveRequest : RetrieveRequest {
    public int MyOtherIntProperty { get; set; }
    public string MyOtherStringProperty { get; set; }
}
```

**Comment by teafmartins** [comment]
I don't think I've explained myself in the best way. 

I want the request (used by [EditLink]) to contain not only the EntityId but also two other columns/fields with their respective values. It should be noted that the EntityId is a sequential number that doesn't allow me to look up the columns/fields I want within Retrieve.
I hope I managed to explain :). Any ideas?

Thanks @VictorTomaili  for your help.

**Comment by VictorTomaili** [comment]
This example can help you 

https://demo.serenity.is/BasicSamples/InlineActionButtons

https://github.com/serenity-is/common-features/blob/master/src/Serenity.Demo.BasicSamples/Modules/Grids/InlineActionButtons/InlineActionButtonsPage.ts
